docs/examples/use_cases/tensorflow/yolov4/src/dali/ops.py

```diff
@@ -0,0 +1,212 @@
+import nvidia.dali as dali
+
+def input(file_root, annotations_file, device_id, num_threads, device, random_shuffle=True):
+    inputs, bboxes, classes = dali.fn.readers.coco(
+        file_root=file_root,
+        annotations_file=annotations_file,
+        ltrb=True,
+        shard_id=device_id,
+        num_shards=num_threads,
+        ratio=True,
+        random_shuffle=random_shuffle
+    )
+    images = dali.fn.decoders.image(inputs, device=device, output_type=dali.types.RGB)
+
+    return images, bboxes, classes
+
+def ltrb_to_xywh(bboxes):
+    Z = dali.types.Constant(0.0)
+    H = dali.types.Constant(0.5)
+    O = dali.types.Constant(1.0)
+    MO = dali.types.Constant(-1.0)
+
+    M = dali.fn.stack(
+            dali.fn.stack(H, Z, H, Z),
+            dali.fn.stack(Z, H, Z, H),
+            dali.fn.stack(MO, Z, O, Z),
+            dali.fn.stack(Z, MO, Z, O)
+        )
+    return dali.fn.coord_transform(bboxes, M=M)
+
+
+def bbox_adjust_xywh(bboxes, shape_x, shape_y, pos_x, pos_y):
+    Z = dali.types.Constant(0.0)
+
+    M = dali.fn.stack(
+            dali.fn.stack(shape_x, Z, Z, Z),
+            dali.fn.stack(Z, shape_y, Z, Z),
+            dali.fn.stack(Z, Z, shape_x, Z),
+            dali.fn.stack(Z, Z, Z, shape_y)
+        )
+
+    T = dali.fn.stack(pos_x, pos_y, Z, Z)
+    return dali.fn.coord_transform(bboxes, M=M, T=T)
+
+def bbox_adjust_ltrb(bboxes, shape_x, shape_y, pos_x, pos_y):
+    sx, sy, ex, ey = pos_x, pos_y, shape_x + pos_x, shape_y + pos_y
+    MT = dali.fn.transforms.crop(
+        to_start=dali.fn.stack(sx, sy, sx, sy),
+        to_end=dali.fn.stack(ex, ey, ex, ey)
+    )
+    return dali.fn.coord_transform(bboxes, MT=MT)
+
+
+# Works for 2D tensors with equal second dimension size
+def select(bool_TL, if_true, if_false):
```

Comment:
```suggestion
def select(predicate, if_value, else_value):
```
So it selects one of two inputs based on the predicate. The shape doesn't have to match, right?

Reply:
And I guess it works only for CPU inputs, right?

Reply:
> So it selects one of two inputs based on the predicate. 
Yes.
The shape must match on axis 1, but doesn't have to match on axis 0. It happens to be a case when selecting whether to use bboxes from original image or from mosaic augmented image, as the number of bboxes can differ between the two.
I believe it should also work for GPU inputs, because it uses only slices and cats.

Reply:
But you use shapes and shape output for GPU input is also on GPU so it won't work.